pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,214 @@
+package comment
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/issue/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/pkg/surveyext"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type CommentOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Edit       func(string) (string, error)
+
+	SelectorArg string
+	Interactive bool
+	InputType   int
+	Body        string
+}
+
+const (
+	inline = iota
+	editor
+	web
+)
+
+func NewCmdComment(f *cmdutil.Factory, runF func(*CommentOptions) error) *cobra.Command {
+	opts := &CommentOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+		Edit: func(editorCommand string) (string, error) {
+			return surveyext.Edit(editorCommand, "*.md", "", f.IOStreams.In, f.IOStreams.Out, f.IOStreams.ErrOut, nil)
+		},
+	}
+
+	var webMode bool
+	var editorMode bool
+
+	cmd := &cobra.Command{
+		Use:   "comment {<number> | <url>}",
+		Short: "Create a new issue comment",
+		Example: heredoc.Doc(`
+			$ gh issue comment 22 --body "I found a bug. Nothing works"
```

Comment:
Should we add more examples?

Reply:
Examples aren't strictly necessary if each of the flags is well-documented. For instance, passing a string to a flag called `--body` should be obvious to most readers and we do not necessarily have to demonstrate it.

Reply:
I like examples! Even if it's well documented it can just be a little easier to understand with them. In this case, I think it's helpful to see you need the flag and can't just pass the comment as an argument alone. But I don't think we need to add more just for the sake of it.